src/bp-activity/bp-activity-filters.php

```diff
@@ -2360,25 +2043,489 @@ function bp_blogs_activity_content_with_read_more( $content, $activity ) {
+	if ( ! empty( $recipients ) ) {
+		$records_updated = '';
+		foreach ( $recipients as $get_parent_id ) {
+			$check_media_migration = bp_activity_get_meta( $get_parent_id->id, 'bp_media_comment_migration', true );
+			if ( 'success' !== $check_media_migration ) {
+				$args = array(
+					'display_comments' => true,
+					'show_hidden'      => true,
+					'sort'             => 'ASC',
+					'activity_ids'     => $get_parent_id->id,
+				);
+				bb_migration_write_log( 'Main Parent Id - ' . $get_parent_id->id );
+				$get_activity_data     = bp_activity_get_specific( $args );
+				$new_seq               = array();
+				$sub_data              = array();
+				$sub_activity_id_store = array();
+				if ( isset( $get_activity_data['activities'] ) ) {
+					foreach ( $get_activity_data['activities'] as $comment_data ) {
+						if ( $comment_data->children ) {
+							$sub_data                              = bb_migration_get_children_data(
+								$sub_activity_id_store,
+								$sub_data,
+								$comment_data->children,
+								$get_parent_id->id
+							);
+							$new_seq[ $comment_data->id ]['child'] = $sub_data['child_array'];
+							$sub_activity_id_store                 = $sub_data['activity_id_store'];
+						}
+					}
+					if ( ! empty( $sub_activity_id_store ) ) {
+						// Remove id which type is activity_update and privacy is media.
+						// Update in meta when migration complete for the root id.
+						bb_migration_remove_activity_id_activity_update_type( $sub_activity_id_store, $get_parent_id->id );
+						bp_activity_update_meta( $get_parent_id->id, 'bp_media_comment_migration', 'success' );
+						/* translators: %s refers to how many numbers running at that process */
+						$records_updated = sprintf( __( '%s media comment migrated successfully.', 'buddyboss' ), number_format_i18n( $offset ) );
+					} else {
+						/* translators: %s refers to how many numbers running at that process */
+						$records_updated = sprintf( __( '%s no media comment available.', 'buddyboss' ), number_format_i18n( $offset ) );
+					}
+				}
+			} else {
+				/* translators: %s refers to how many numbers running at that process */
+				$records_updated = sprintf( __( '%s media already migrated.', 'buddyboss' ), number_format_i18n( $offset ) );
+			}
+			$offset ++;
+		}
+		bb_migration_write_log( 'offset - ' . $offset );
+		bb_migration_write_log( '$recipients_count_row_data->ids - ' . $recipients_count_row_data->ids );
+		if ( 1 === (int) $recipients_count_row_data->ids || (int) $offset === (int) $recipients_count_row_data->ids ) {
+			return array(
+				'status'  => 1,
+				'message' => __( 'media comment migration complete!', 'buddyboss' ),
+			);
+		} else {
+			return array(
+				'status'  => 'running',
+				'offset'  => $offset,
+				'records' => $records_updated,
+			);
+		}
+	} else {
+		return array(
+			'status'  => 1,
+			'message' => __( 'media comment migration complete!', 'buddyboss' ),
+		);
+	}
+}
+/**
```

Comment:
add empty line between two functions and please add proper commenting on each important line of codes what it does and describe it.